lib/helpers/BasePage.ts

```diff
@@ -0,0 +1,617 @@
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    await locator.click({ button: 'right', force: options?.force });
+  }
+
+  /**
+   * Clicks an element using JavaScript (bypasses actionability checks).
+   * Use when standard click doesn't work due to overlapping elements.
+   * @param locator - The element to click
+   */
+  async forceClick(locator: Locator): Promise<void> {
+    await locator.evaluate((el: HTMLElement) => el.click());
+  }
+
+  /**
+   * Enters text into an input field after clearing it.
+   * Verifies element visibility before interaction.
+   * @param locator - The input element
+   * @param text - The text to enter
+   * @param options - Optional configuration
+   */
+  async enterText(
+    locator: Locator,
+    text: string,
+    options?: { clearFirst?: boolean; verify?: boolean; timeout?: number }
+  ): Promise<void> {
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    if (options?.clearFirst !== false) {
+      await locator.clear();
+    }
+    await locator.fill(text);
+    if (options?.verify) {
+      await expect(locator).toHaveValue(text);
+    }
+  }
+
+  /**
+   * Types text character by character (simulates real typing).
+   * Useful when fill() doesn't trigger necessary events.
+   * @param locator - The input element
+   * @param text - The text to type
+   * @param options - Optional configuration
+   */
+  async typeText(
+    locator: Locator,
+    text: string,
+    options?: { clearFirst?: boolean; delay?: number; timeout?: number }
+  ): Promise<void> {
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    if (options?.clearFirst !== false) {
+      await locator.clear();
+    }
+    await locator.pressSequentially(text, { delay: options?.delay ?? 50 });
+  }
+
+  /**
+   * Clears an input field.
+   * @param locator - The input element to clear
+   */
+  async clearText(locator: Locator, options?: { timeout?: number }): Promise<void> {
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    await locator.clear();
+  }
+
+  /**
+   * Presses a keyboard key while focused on an element.
+   * @param locator - The element to focus
+   * @param key - The key to press (e.g., 'Enter', 'Tab', 'Escape')
+   */
+  async pressKey(locator: Locator, key: string, options?: { timeout?: number }): Promise<void> {
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    await locator.press(key);
+  }
+
+  /**
+   * Selects an option from a dropdown by value.
+   * @param locator - The select element
+   * @param value - The option value to select
+   */
+  async selectByValue(locator: Locator, value: string, options?: { timeout?: number }): Promise<void> {
+    await expect(locator).toBeVisible({ timeout: options?.timeout ?? 5000 });
+    await locator.selectOption({ value });
```

Comment:
```suggestion
    await expect(locator).toBeVisible({timeout: options?.timeout ?? 5000});
    await locator.selectOption({value});
```

There is more places the indentation should be fixed